### PR TITLE
complete getCommentsByArticleId and implemented all the necessary tes…

### PR DESCRIPTION
**Pull request by brunoFernandes21**
…ts, including 404 and 400

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -121,3 +121,40 @@ describe("handles all bad paths", () => {
...
+
+describe("GET /api/articles/:article_id/comments", () => {
+  test("200: should respond with an array of comments for the given article_id", () => {
+    return request(app)
+    .get("/api/articles/1/comments")
+    .expect(200)
+    .then(( { body }) => {
+      const { comments } = body
+      expect(comments).toHaveLength(11)
+      expect(comments).toBeSortedBy("created_at", {descending: true})
```
Comment (mezzDavies) [inline __tests__/app.test.js:133]: Good stuff on remembering and implementing this 👍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -121,3 +121,40 @@ describe("handles all bad paths", () => {
...
+    return request(app)
+    .get("/api/articles/99999999/comments")
+    .expect(404)
+    .then(( { body }) => {
+      expect(body.msg).toBe("Not Found")
+    })
+  })
+  test("400: should respond with Bad request message when passed an invalid article_id", () => {
+    return request(app)
+    .get("/api/articles/strawberries/comments")
```
Comment (mezzDavies) [inline __tests__/app.test.js:163]: 🍓🍓🍓🍓🍓🍓🍓🍓

File: __tests__/app.test.js [hunk 3]
```diff
@@ -121,3 +121,40 @@ describe("handles all bad paths", () => {
...
+      comments.forEach((comment) => {
+        expect(comment.article_id).toBe(1)
+        expect(comment).toHaveProperty("author", expect.any(String))
+        expect(comment).toHaveProperty("body", expect.any(String))
+        expect(comment).toHaveProperty("article_id", expect.any(Number))
+        expect(comment).toHaveProperty("created_at", expect.any(String))
+        expect(comment).toHaveProperty("votes", expect.any(Number))
+      })
+    })
+  })
```
Comment (mezzDavies) [inline __tests__/app.test.js:143]: There is one other happy path test - what if the article exists but it has no comments? We should return the empty array as these are the comments for that article - none. You will need to find a way to stop the 404 error being sent in this circumstance.

**Review by mezzDavies — CHANGES_REQUESTED** [review]
Good stuff Bruno! Just one further test you need to think about though - see my comment below. Please resubmit when you have tested and implemented the required code 🙂

File: __tests__/app.test.js [hunk 4]
```diff
@@ -121,3 +121,49 @@ describe("handles all bad paths", () => {
...
+  test("200: should respond with an empty array if article_id exists but has no comments", () => {
+    return request(app)
+    .get("/api/articles/2/comments")
+    .expect(200)
+    .then(( { body}) => {
+      const { comments } = body
+      expect(comments).toHaveLength(0)
+    })
```
Comment (NRMSMITH) [inline __tests__/app.test.js:151]: Excellent implementation of feedback Bruno! 💘

**Review by NRMSMITH — APPROVED** [review]
Well done! 💘

**Pull request merged**